File: OLD SNACKS/JAVASCRIPT/semicolonCheckoutSystem.js
const prompt = require("Prompt-sync")();
let productDetails = [] 
let subtotal = 0.0

function inputProductDetails() {
	while (true) {
		const productName = prompt("What did the customer buy:   ");
		const quantity = prompt("How many pieces:   ");
		const pricePerUnit = parseFloat(prompt("How many per unit:   "));

		const totalPrice = quantity * pricePerUnit;

		productDetails.push(`${productName}\t${quantity}\t${pricePerUnit.toFixed(2)}\t${totalPrice.toFixed(2)}`);
		subtotal += totalPrice;

		const response = prompt("Add more items? (yes/no):    ");
		if (response.toLowerCase() !=="yes") {
			break;
		}
	}
}

function calculateDiscount(discountPercentage) {
	return (discountPercentage / 100) * subtotal;
}

function calculateVAT(totalAfterDiscount) {
	return 0.075 * totalAfterDiscount;
}

function displayInvoice(customerName, cashierName) {
	console.log("\nSemicolon Stores");
	console.log("Main Branch");
	console.log("Location: 312	, HERBERT MACULAY WAY, SABO YABA, LAGOS.");
	console.log("TEL: 03293828343");
	console.log("Date:   ", getCurrentDate());
	console.log("Cashier:   ", cashierName);
	console.log("Customer:   ", customerName);
	console.log("===========================================");
	console.log("ITEM\t\tQTY\tPRICE\t\TOTAL(NGN)");
	
	productDetails.forEach(detail => {
		console.log(detail);
	});
	
	console.log("------------------------------------------------------------------------------");
	console.log(`subtotal:   ${subtotal.toFixed(2)}`);
	const discountPercentage = parseFloat(prompt("How much discount will he/she get:   "));
	const discount = calculateDiscount(discountPercentage);
	console.log(`Discount:   ${discount.toFixed(2)}`);
	const totalAfterDiscount = subtotal - discount;
	const vat = calculateVAT(totalAfterDiscount);
	console.log(`VAT (7.5%):  ${vat.toFixed(2)}`);
	const total = totalAfterDiscount + vat;
	console.log("===========================================");
	console.log(`Total:  ${total.toFixed(2)} (after applying discount and VAT)`);
	console.log("===========================================");
	return total;
}

function printReceipt(customerName, cashierName, total) {
	const amountPaid = parseFloat(prompt("How much did the customer pay:   "));
	console.log("\n=================RECEIPT====================");
	console.log("Thank you for shopping with us!");
	console.log("Customer Name:  ", customerName);
	console.log("Cashier:   ", cashierName);
	console.log("Total Amount:   ", total.toFixed(2));
	console.log("Amount Paid    ", amountPaid.toFixed(2));
	const change = amountPaid - total;
	if (change < 0){
		console.log(`Amount Due:   ${(-change).toFixed(2)}`);
	} else {
		console.log(`change:   ${change.toFixed(2)}`);
	}
	console.log("============================================");
}

function getCurrentDate() {
	const date = new Date();
	return date.toISOString().split('T')[0];
}

const customerName = prompt("What is the Customer's Name:   ");
const cashierName = prompt("What is the Cashier's Name:    ");

inputProductDetails();
const total = displayInvoice(customerName, cashierName);
printReceipt(customerName, cashierName, total);

File: OLD SNACKS/PYTHON/semicolon_checkout_system.py
from datetime import datetime

product_details = []
subtotal = 0.0

def input_product_details():
	global subtotal
	while True:
		product_name = input("What did the customer buy:   ")
		quantity = int(input("How many pieces:   "))
		price_per_unit = float(input("How much per unit:   "))
		
		total_price = quantity * price_per_unit
		product_details.append(f"{product_name}\t{quantity}\t{price_per_unit: .2f}\t{total_price: .2f}")
		subtotal += total_price


		response = input("Add more itmes? (yes/n0):   ")
		if response.lower() != "yes":
			break

def calculate_discount(discount_percentage):
	return (discount_percentage / 100) * subtotal

def calculate_vat(total_after_discount):
	return 0.075 * total_after_discount

def display_invoice(customer_name, cashier_name):
	print("\nSemicolon Stores")
	print("Main branch")
	print("Location: 312, HERBERT MACULAY WAY, SABO YABA, LAGOS.")
	print("TEL:   03293828343")
	print("DATE:  ", get_current_date())
	print("Cashier:   ", cashier_name)
	print("Customer Name:   ", customer_name)
	print("===========================================")
	print("ITEM\t\tPRICE\t\tTOTAL(NGN)")

	for detail in product_details:
		print(detail)

	print("------------------------------------------------------------------------------")
	print(f"Subtotal:   {subtotal: .2f}")
	discount_percentage = float(input("how much discount will he get:   "))
	discount = calculate_discount(discount_percentage)
	print(f"Discount:   {discount: .2f}")
	total_after_discount = subtotal - discount
	vat = calculate_vat(total_after_discount)
	print(f"VAT (7.5%):   {vat: .2f}")
	total = total_after_discount + vat
	print("==========================================")
	print(f"Total:   {total: .2f} (after applying discount and VAT)")
	print("==========================================")
	return total

def print_receipt(customer_name, cashier_name, total):
	amount_paid = float(input("How much did the customer pay:   "))
	print("\n=============== RECEIPT ===================")
	print("Thank you for shopping with us!")
	print("Customer name:   ", customer_name)
	print("Cashier's name:    ", cashier_name)
	print("Total Amount:    ", total)
	print("Amount Paid:    ", amount_paid)
	change = amount_paid - total
	if change < 0:
		print(f"Amount Due:   {-change: .2f}")
	else: 
		print(f"Change: {change: .2f}")
	print("===========================================")

def get_current_date():
	return datetime.now().strftime("%y-%m-%d")

if __name__ == "__main__":
	customer_name = input("What is the customer's name:    ")
	cashier_name = input("What is the cashier name:    ")
	
	input_product_details()
	total = display_invoice(customer_name, cashier_name)
	print_receipt(customer_name, cashier_name, total)
	
	